gems/pending/disk/modules/AzureBlobDisk.rb

```diff
@@ -0,0 +1,91 @@
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module AzureBlobDisk
+  # The maximum read length that supports MD5 return.
+  MAX_READ_LEN = 1024 * 1024 * 4
+
+  def self.new(svc, blob_uri, dInfo = nil)
+    d_info = dInfo || OpenStruct.new
+    d_info.storage_acct_svc = svc
+    d_info.blob_uri         = blob_uri
+    d_info.fileName         = blob_uri
+
+    MiqDisk.new(self, d_info, 0)
+  end
+
+  def d_init
+    @diskType         = "azure-blob"
+    @blockSize        = 512
+    @blob_uri         = @dInfo.blob_uri
```

Comment:
Some of the existing variables are referenced by the super class, and I didn't want to change that class in this PR. The new variables are only used by this class, so I made them snake case. Ultimately, my goal is to change them all to snake case - probably when we re-work this disk area.